.github/workflows/push_pr_checks_tests.yml

```diff
@@ -0,0 +1,440 @@
+          toolchain: ${{ env.RUST_VERSION }}
+
+      - uses: actions/cache/restore@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}-test-dev
+
+      - name: Run onHost root-required integration-tests only
+        run: make -C agent-control test/onhost/root/integration
+
+  embedded-downloader-tests:
+    name: Unit test for embedded downloader
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: "build/embedded/go.mod"
+      - name: run tests for embedded downloader
+        run: make -C build/embedded test
+
+  k8s-integration-tests:
+    name: K8s integration tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        group: [ part1, part2, flux-self-update ]
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Minikube
+        uses: manusa/actions-setup-minikube@v2.14.0
+        with:
+          minikube version: v1.35.0
+          kubernetes version: "v1.30.14"
+          driver: docker
+          github token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Tilt
+        run: |
+          curl -fsSL https://raw.githubusercontent.com/tilt-dev/tilt/master/scripts/install.sh | bash
+
+      - name: Obtain Rust version from project
+        run: |
+          RUST_VERSION=$(grep "rust-version" Cargo.toml | cut -d "=" -f2 | tr -d "[:space:]")
+          echo "RUST_VERSION=${RUST_VERSION}" >> $GITHUB_ENV
+
+      - name: Install Rust ${{ env.RUST_VERSION }}
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_VERSION }}
+          targets: >-
+            x86_64-unknown-linux-musl
+
+      - uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}-test-dev
+
+      - name: Install Zig
+        uses: mlugg/setup-zig@v2
+
+      - name: Install cargo-zigbuild
+        run: |
+          cargo install --locked cargo-zigbuild --force
+
+      - name: Run k8s integration tests
+        env:
+          # Tilt configs
+          ARCH: amd64
+        run: make -C agent-control test/k8s/integration-${{ matrix.group }}
+
+  k8s-e2e-tests:
+    uses: ./.github/workflows/component_k8s_e2e.yml
+    with:
+      scenarios: '["infra"]'
+    secrets:
+      E2E_ACCOUNT_ID: ${{ secrets.AC_PROD_E2E_ACCOUNT_ID }}
+      E2E_API_KEY: ${{ secrets.AC_PROD_E2E_API_KEY }}
+      E2E_LICENSE_KEY: ${{ secrets.AC_PROD_E2E_LICENSE_KEY }}
+
+  check-all-green:
+    name: 🟢 All required checks and tests pass
```

Comment:
Do 'Github settings' require the job key or the job name when configuring required jobs? I wonder if it will work well with the emoji 🤔